source/grid/grid_tools.cc

```diff
@@ -2103,6 +2103,59 @@ namespace GridTools
     return output_tuple;
   }
 
+  template <int spacedim>
+#ifndef DOXYGEN
+  std::tuple<std::map<unsigned int, std::vector<unsigned int>>,
+             std::map<unsigned int, unsigned int>,
+             std::map<unsigned int, std::vector<unsigned int>>>
+#else
+  return_type
+#endif
+  guess_point_owner(
+    const RTree<std::pair<BoundingBox<spacedim>, unsigned int>> &covering_rtree,
+    const std::vector<Point<spacedim>> &                         points)
+  {
+    std::map<unsigned int, std::vector<unsigned int>> point_owners;
+    std::map<unsigned int, unsigned int>              map_owners_found;
+    std::map<unsigned int, std::vector<unsigned int>> map_owners_guessed;
```

Comment:
I guess it is too late to change this (since the other `guess_point_user` uses this type), but why are these `std::map`s? If they are indexed by point numbers shouldn't we just use `std::vector`?

Reply:
I made that choice for a reason (even though maybe it is not good enough): the idea is that if I use, say 1000 processes, a process probably "owns" points only inside a few of them (if not, then all the messaging will be the problem anyway). Then I don't want to store/reserve three very big vectors of vectors knowing that most of that space won't be used.

Reply:
Oh, I see: I thought `map_owners_found` and `map_owners_guessed` were the same in the copying check below, but they are not.